docs/source/tutorial/convnet.rst

```diff
@@ -0,0 +1,428 @@
+at the source.
+
+Chainer is flexible, so we can write our original network in many different
+ways. It would give users an intuitive way to design brand-new and complex
+models.
+
+VGG16
+'''''
+
+Next, let's write more large models like VGG16 [Simonyan14]_ in Chainer.
+When you write a large network consisted of several building block
+networks, :class:`~chainer.ChainList` is useful. First, let's see how to write
+a VGG16 model:
+
+.. doctest::
+
+    class VGG16(chainer.ChainList):
+
+        def __init__(self):
+            w = chainer.initializers.HeNormal()
+            super(VGG16, self).__init__(
+                VGGBlock(64),
+                VGGBlock(128),
+                VGGBlock(256, 3),
+                VGGBlock(512, 3),
+                VGGBlock(512, 3, True))
+            self.train = True
+
+        def __call__(self, x):
+            for f in self.children():
+                x = f(x, self.train)
+            if self.train:
+                return x
+            return F.softmax(x)
+
+    class VGGBlock(chainer.Chain):
+
+        def __init__(self, n_channels, n_convs=2, fc=False):
+            w = chainer.initializers.HeNormal()
+            super(VGG16Block, self).__init__(
+                conv1=L.Convolution2D(None, n_channels, 3, 1, 1, initialW=w),
+                conv2=L.Convolution2D(
+                    n_channels, n_channels, 3, 1, 1, initialW=w))
+            if n_convs == 3:
+                self.add_link('conv3', L.Convolution2D(
+                    n_channels, n_channels, 3, 1, 1, initialW=w))
+            if fc:
+                self.add_link('fc4', L.Linear(None, 4096, initialW=w))
+                self.add_link('fc5', L.Linear(4096, 4096, initialW=w))
+                self.add_link('fc6', L.Linear(4096, 1000, initialW=w))
+
+            self.n_convs = n_convs
+            self.fc = fc
+
+        def __call__(self, x, train):
+            h = F.relu(self.conv1(x))
+            h = F.relu(self.conv2(h))
+            if self.n_convs == 3:
+                h = F.relu(self.conv3(h))
+            h = F.max_pooling_2d(h, 2, 2)
+            if self.fc:
+                h = F.dropout(F.relu(self.fc4(h)), train=train)
+                h = F.dropout(F.relu(self.fc5(h)), train=train)
+                h = self.fc6(h)
+            return h
+
+That's it. VGG16 is a model which won the 1st place in
+`classification + localization task at ILSVRC 2014 <http://www.image-net.org/challenges/LSVRC/2014/results#clsloc>`_,
+and since then, became one of standard models for many different tasks as a
+pre-trained model. This has 16-layers, so it's called "VGG-16", but we can
+write this model without writing all layers independently. Because this model
+is consisted of several building blocks that have the same architecture,
+so that we can build the whole network by re-using the building block
+definition. Each part of the network is consisted of 2 or 3 convolutional
```

Comment:
Since this model consists of several building blocks that have the same architecture, we can build the whole network by re-using the building block definition.